### PR TITLE
Fix for GUI clipping when bug when parent container is scaled

### PR DESCRIPTION
**Pull request by msDestiny14**
Also tested with this playground as well as the GUI editor. 
https://www.babylonjs-playground.com/debug.html#F6XWSQ#2

### CODE REVIEW

File: gui/src/2D/controls/control.ts [hunk 1]
```diff
@@ -1617,24 +1617,32 @@ export class Control {
     }
 
     protected _evaluateClippingState(parentMeasure: Measure) {
+        let evaluatedMeasure = new Measure(0, 0, 0, 0);
```
Comment (sebavan) [inline gui/src/2D/controls/control.ts:1620]: We should not allocate as much, you could keep the instance of measure in cache and reuse it. Also, Nit: it should be a const._evaluateClippingState

File: gui/src/2D/controls/control.ts [hunk 2]
```diff
@@ -1617,24 +1617,32 @@ export class Control {
...
+        this._currentMeasure.transformToRef(this._transformMatrix, evaluatedMeasure);
+
+        let evaluatedParentMeasure = new Measure(0, 0, 0, 0);
```
Comment (sebavan) [inline gui/src/2D/controls/control.ts:1623]: We should not allocate as much, you could keep the instance of measure in cache and reuse it. Also, Nit: it should be a const._evaluateClippingState

File: gui/src/2D/controls/control.ts [hunk 3]
```diff
@@ -1617,24 +1617,32 @@ export class Control {
...
+        const parentMatrix = this.parent?._transformMatrix;
```
Comment (sebavan) [inline gui/src/2D/controls/control.ts:1624]: You should move the 3 lines in the if block below as it already checks for parent nullity. preventing the need for the ? and parentMatrix test.

**Review by sebavan — CHANGES_REQUESTED** [review]
I ll let @RaananW do the final pass on Monday as he was familiar with the issue.

**Comment by RaananW** [comment]
I have no other comments - Seb nailed them all. 

Regarding updating the transformMatrix - if it is working without it then it's all fine. I didn't change the code, I only debugged using the console, so I might have missed something there.

File: gui/src/2D/controls/control.ts (hunk 1 above)
Comment (msDestiny14) [reply gui/src/2D/controls/control.ts:1620]: Not sure what you mean by this: const._evaluateClippingState?

**Pull request merged**